Data/models/CRUDProtocols.swift

```diff
@@ -82,12 +78,15 @@ public extension Readable where Self: NSManagedObject {
         return nil
     }
     
-    static func first(where predicate: NSPredicate?, context: NSManagedObjectContext = DataController.viewContext) -> Self? {
-        return all(where: predicate, fetchLimit: 1, context: context)?.first
+    static func first(where predicate: NSPredicate? = nil, sortDescriptors: [NSSortDescriptor]? = nil,
+                      context: NSManagedObjectContext? = nil) -> Self? {
+        let context = context ?? DataController.viewContext
```

Comment:
`first` is public, `viewContext` is internal, and it can't be used as a default parameter.

I could make `viewContext` public, but sure what is better. Thinking about architecture I wanted to remove all direct references to `DataController` from `Client`

Reply:
Unfortunate, but understandable.